vm/devices/firmware/uefi_specs/src/hyperv/advanced_logger.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Types and constants defined by Project Mu's Advanced Logger package,
+//! used in the Hyper-V UEFI firmware.
+
+#![warn(missing_docs)]
+
+use crate::uefi::time::EFI_TIME;
+use zerocopy::FromBytes;
+use zerocopy::Immutable;
+use zerocopy::KnownLayout;
+
+/// Advanced Logger Info signature
+pub const SIG_HEADER: u32 = u32::from_le_bytes(*b"ALOG");
+
+/// Advanced Logger Entry signature
+pub const SIG_ENTRY: u32 = u32::from_le_bytes(*b"ALM2");
+
+/// UEFI Advanced Logger Info Header, which is shared
+/// with the Advanced Logger Package in UEFI. The entries
+/// live right after.
+#[repr(C, packed)]
```

Comment:
Why is this packed? It looks like it has natural alignment for all fields. Marking it packed just makes more work for yourself since you can't take references to fields inside it.